Assets/Scripts/Framework/Gameplay/LevelTracker.cs

```diff
@@ -0,0 +1,208 @@
+
+        [SerializeField]
+        private int _succesfullDeliveryBonus = 5;
+
+        [SerializeField]
+        private int _incorrectDeliveryPenalty = 5;
+
+        [SerializeField]
+        private int _totalScore;
+
+        [SerializeField]
+        private int _currentScore;
+
+        [SerializeField]
+        private GameObject _packagePrefab;
+
+        [SerializeField]
+        private TrainStation[] _allStations;
+
+        /// <summary>
+        /// An action that broadcasts when a package is delivered and the amount of points that is added.
+        /// </summary>
+        public Action<int> OnPackageDelivered;
+
+        private static readonly char[] CHARACTERS = "QWERTYUIOPASDFGHJKLZXCVBNM".ToCharArray();
+
+        public override void OnStartServer()
+        {
+            base.OnStartServer();
+
+            SelectLevelToLoad(0);
+        }
+
+        /// <summary>
+        /// A function that is used to load a new level.
+        /// </summary>
+        /// <param name="index">The level ID.</param>
+        [Server]
+        private void SelectLevelToLoad(int index)
+        {
+            _currentScore = 0;
+
+            StationLevelData[] levelData = _levels.levels[index].StationLevelData;
+            List<string> labels = GenerateLabelsForStations(levelData);
+
+            List<Transform> usedSpawns = SpawnDeliveries(levelData, labels);
+
+            List<Transform> freeSpawns = GetAllFreeSpawns(usedSpawns);
+
+            SpawnFakeDeliveries(freeSpawns, labels);
+        }
+
+        private List<string> GenerateLabelsForStations(StationLevelData[] stationData)
+        {
+            System.Random random = new System.Random();
+            List<string> labels = new();
+
+            for (int i = 0; i < stationData.Length; i++)
+            {
+                TrainStation targetStation = _allStations[i];
+                string label = string.Empty;
+
+                while (label == string.Empty || labels.Contains(label))
+                {
+                    label
+                        = CHARACTERS[random.Next(0, CHARACTERS.Length)].ToString()
+                        + CHARACTERS[random.Next(0, CHARACTERS.Length)].ToString();
+                }
+
+                labels.Add(label);
+                targetStation.UpdateLabelAndID(label, i);
+            }
+
+            return labels;
+        }
+
+        private List<Transform> SpawnDeliveries(StationLevelData[] levelData, List<string> labels)
+        {
+            List<Transform> usedSpawns = new();
+
+            for (int i = levelData.Length - 1; i >= 0; i--)
+            {
+                List<Transform> availableSpawns = GetAvailableSpawnsForStation(i, usedSpawns);
+                int amountToSpawn = levelData[i].minDeliverablePackages;
+
+                while (amountToSpawn > 0 && availableSpawns.Count > 0)
+                {
+                    availableSpawns.Shuffle();
+                    Transform targetSpawn = availableSpawns[0];
```

Comment:
Just use a random instead of `Shuffle()`.

Reply:
This is easier.

Reply:
It's very easy to implement and the relative performance difference is big. You can just use Unitys Random class instead of the `System.Random` class. That will make it even more clean.